cpp/generate_issn_to_titles_and_ppns_map.cc

```diff
@@ -67,7 +69,35 @@ auto FilterOutInvalidISSNs(const std::set<std::string> &unvalidated_issns) {
 }
 
 
-unsigned ProcessZederAndWriteMapFile(File * const output, const Zeder::SimpleZeder &zeder) {
+struct JournalDescriptor {
+    std::set<std::string> print_issn_;
+    std::string online_issn_;
+    std::string print_ppn_;
+    std::string online_ppn_;
+    std::string title_;
+public:
+    JournalDescriptor(const std::string &print_issn, const std::string &online_issn, const std::string &print_ppn,
+                      const std::string &online_ppn, const std::string &title)
+        : print_issn_(print_issn), online_issn_(online_issn), print_ppn_(print_ppn), online_ppn_(online_ppn),
+          title_(title) { }
+    void serialize(File * const output) const;
+};
+
+
+void JournalDescriptor::serialize(File * const output) const {
+    if (not print_issn_.empty()) {
+        (*output) << print_issn_ << ':' << EscapeColons(title_) << ':' << (online_issn_.empty() ? print_issn_ : online_issn_)
+                  << (online_ppn_.empty() ? print_ppn_ : online_ppn_) << '\n';
+    }
+
+    if (not online_issn_.empty() and not online_ppn_.empty())
+        (*output) << online_issn_ << ':' << EscapeColons(title_) << ':' << online_ppn_ << '\n';
```

Comment:
Title might be doubled if print issn/ppn and online issn/ppn is given. Is this really necessary?